addons/account_edi/models/account_move.py

```diff
@@ -450,6 +453,36 @@ def _process_attachments_for_template_post(self, mail_template):
                 move_result.setdefault('attachments', []).extend(edi_attachments.get('attachments', []))
         return result
 
+    ####################################################
+    # Export Electronic Document
+    ####################################################
+
+    def action_download_electronic_invoice(self):
+        ids = [invoice.id for invoice in self if invoice.edi_document_ids]
+        if not ids:
+            return
+        url = '/web/edi_document/download?ids=' + ','.join(
+            str(id)
+            for id in ids
+        )
+
+        return {
+            'type': 'ir.actions.act_url',
+            'url': url,
+            'target': 'new',
+        }
+
+    def _create_zipped(self):
+        buffer = io.BytesIO()
+        zipfile_obj = zipfile.ZipFile(buffer, 'w')
+        for invoice in self:
+            for document in invoice.edi_document_ids:
+                if document.state in ['sent', 'cancelled']:
+                    name = (invoice.name + '_' + document.edi_format_name).replace('/', '_').replace(' ', '_').replace('.', '_').replace('(', '').replace(')', '') + '.xml'
+                    zipfile_obj.writestr(name, base64.b64decode(document.attachment_id.datas), compress_type=zipfile.ZIP_DEFLATED)
+        zipfile_obj.close()
+        buffer.seek(0)
+        return buffer
```

Comment:
With the `with` statement and the change in the controller, I think this works:
```suggestion
        return buffer.getvalue()
```